File: 2144-minimum-cost-of-buying-candies-with-discount/2144-minimum-cost-of-buying-candies-with-discount.py
class Solution:
    def minimumCost(self, cost: List[int]) -> int:

        c=0
        s=sorted(cost,reverse=True)
        n=len(cost)

        
        for i in range(0,n):
            if(i%3!=2):
                c+=s[i]
        return c


        

File: 2144-minimum-cost-of-buying-candies-with-discount/2144-minimum-cost-of-buying-candies-with-discount.java
class Solution {
    public int minimumCost(int[] cost) {
        if(cost.length==1){
            return cost[0];
        }
        Arrays.sort(cost);
        int i=cost.length-1;
        int sum=0;
        while(i>=0){
            if(i-1>=0){
            sum+=cost[i]+cost[i-1];
            }
            else{
                sum+=cost[i];

            }
            i-=3;
        }
        return sum;
        
    }
}